### PR TITLE
Allow escaping a dollar sign when expanding external labels

### LINKED ISSUE #10122: config: Need escaping to avoid environment variable expansion in external labels
If I set the feature flag `--enable-feature=expand-external-labels`, I get the nice and sweet expansion of environment variables in configured `external_labels`.

Should I require a literal `$` in one of my label values, I can simply not set the feature flag.

But what if I want the expansion in some label values but require a literal `$` in others?

Also, we would like to promote the feature to stable and get rid of the feature flag eventually (which we can only do with v3 because it is a breaking change). Once we have done so, we definitely need a way to specify a literal `$`.

I propose to use the same escaping as in Bash, i.e. a literal `$` can be created via `\$`.

(I'm actually surprised that https://pkg.go.dev/os#Expand doesn't provide any escaping out of the box.)

### PR DESCRIPTION
**Pull request by fpetkovski**
There is currently no mechanism to natively escape a dollar sign
in the os.Expand function. As a workaround, this commit modifies
the external label expansion logic to treat a double dollar ($$)
as a mechanism for escaping the dollar character.

Fixes #10122.

Signed-off-by: fpetkovski <filip.petkovsky@gmail.com>

### CODE REVIEW

**Comment by roidelapluie** [comment]
We should test FOO$$BAR as well

**Comment by fpetkovski** [comment]
@roidelapluie added a test for that case.

**Comment by beorn7** [comment]
Fixes #10122.

**Review by LeviHarrison — APPROVED** [review]
Thanks!

File: docs/feature_flags.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ range vector selectors, and subqueries. More details can be found [here](queryin
 Replace `${var}` or `$var` in the [`external_labels`](configuration/configuration.md#configuration-file)
 values according to the values of the current environment variables. References
 to undefined variables are replaced by the empty string.
+The $ character can be escaped by using $$.
```
Comment (LeviHarrison) [inline docs/feature_flags.md:28]: Might be good to put these in backticks.

```suggestion
The `$` character can be escaped by using `$$`.
```

**Review by roidelapluie — APPROVED** [review]

**Comment by roidelapluie** [comment]
Thanks for your PR. For legal reasons, we require that all commits are signed with a [DCO](https://developercertificate.org/) before we can merge them. See [this blog post](https://www.cncf.io/blog/2016/11/08/cloud-native-software-can-trust/) for considerations around this.

This means that the last line of your commit message should read like:

`Signed-Off-By: Your Name <your@email.address>`

If you are using GitHub through the web interface, it's quickest to close this PR and open a new one with the appropriate line.

If you are using Git on the command line, it is probably quickest to amend and force push. You can do that with
````
git commit --amend --signoff
git push -f $remote $remote_branch_for_pr
````

As always, be careful when force-pushing.

**Comment by fpetkovski** [comment]
Ah, the DCO was not added to the suggestion I committed through the browser. It should be there now.

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**